### ISSUE TITLE
Second uv_pipe_accept() returns EBADF on Windows

### ISSUE THREAD

**Issue by brimworks** [libuv/libuv#485]
Originally I filed this bug with the luvit team:

https://github.com/luvit/luv/issues/177

...but I have tracked the bug down to libuv. To reproduce this issue you can grab the luvi source code and build it (on Windows) as such:

```bash
git clone https://github.com/luvit/luvi.git
cd luvi
make tiny
make
```

That will produce a binary in ```build\Release\luvi.exe```. If you download this minimal code snippet you can reproduce the error:

https://gist.github.com/brimworks/c129c707044359d30ed7

To run that code, place it in ```bug\main.lua``` and run this command:

```build\Release\luvi.exe bug```

More details on the exact problem are here:

https://github.com/libuv/libuv/pull/400

Basically, I've narrowed it down to this change (just the change to ```src\win\pipe.c```):

https://github.com/libuv/libuv/commit/fb5df542ace7d6735a9e8e09d7d8bb6b8abdc459
